src/test/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/service/S3ServiceTest.kt

```diff
@@ -43,32 +54,56 @@ internal class S3ServiceTest {
     }
-    whenever(amazonS3Client.putObject(eq("bucket-name"), argForWhich { startsWith("cp/RESULT/UNKNOWN_COURT/") }, eq("message")))
-      .thenReturn(putResult)
+
+    whenever(amazonS3Client.putObject(any<PutObjectRequest>(), any<AsyncRequestBody>())).thenReturn(CompletableFuture.completedFuture(putResult))
 
     val eTag = s3Service.uploadMessage("/hearing/$UUID/result", "message")
 
-    verify(amazonS3Client).putObject(eq("bucket-name"), argForWhich { startsWith("cp/RESULT/UNKNOWN_COURT/") }, eq("message"))
+    verify(amazonS3Client).putObject(putObjectRequestCaptor.capture(), requestBodyCaptor.capture())
+    assertThat(putObjectRequestCaptor.value.bucket()).isEqualTo("bucket-name")
+    assertThat(putObjectRequestCaptor.value.key()).startsWith("cp/RESULT/UNKNOWN_COURT/")
+    assertThat(requestBodyCaptor.value.contentLength().get()).isEqualTo(7)
+
+    // Check the content of the body
+    val capturedBody: AsyncRequestBody = requestBodyCaptor.value
+    val byteBuffer: ByteBuffer = ByteBuffer.allocate(1024)
+    capturedBody.subscribe(Consumer { byteBuffer.put(it) })
+    byteBuffer.flip()
+    val contentOfBody: String = Charsets.UTF_8.decode(byteBuffer).toString()
+
+    assertThat(contentOfBody).isEqualTo("message")
+
     verifyNoMoreInteractions(amazonS3Client)
     assertThat(eTag).isEqualTo("ETAG")
   }
 
   @Test
   fun `given normal JSON input then message is uploaded as file`() {
-    val putResult: PutObjectResult = PutObjectResult().apply {
-      expirationTime = Date()
-      eTag = "ETAG"
+    val putResult: PutObjectResponse = PutObjectRequest.builder().build().let {
+      PutObjectResponse.builder().expiration(Date().toString()).eTag("ETAG").build()
     }
-    whenever(amazonS3Client.putObject(eq("bucket-name"), argForWhich { startsWith("cp/CONFIRM_UPDATE/B10JQ00/") }, any<String>()))
-      .thenReturn(putResult)
+
+    whenever(amazonS3Client.putObject(any<PutObjectRequest>(), any<AsyncRequestBody>())).thenReturn(CompletableFuture.completedFuture(putResult))
 
     val eTag = s3Service.uploadMessage("/hearing/$UUID", minimalJson)
 
-    verify(amazonS3Client).putObject(eq("bucket-name"), argForWhich { startsWith("cp/CONFIRM_UPDATE/B10JQ00/") }, any<String>())
+    verify(amazonS3Client).putObject(putObjectRequestCaptor.capture(), requestBodyCaptor.capture())
+    assertThat(putObjectRequestCaptor.value.bucket()).isEqualTo("bucket-name")
+    assertThat(putObjectRequestCaptor.value.key()).startsWith("cp/CONFIRM_UPDATE/B10JQ00/")
+    assertThat(requestBodyCaptor.value.contentLength().get()).isEqualTo(4655L)
+
+    // Check the content of the body
+    val capturedBody: AsyncRequestBody = requestBodyCaptor.value
+    val byteBuffer: ByteBuffer = ByteBuffer.allocate(4655)
+    capturedBody.subscribe(Consumer { byteBuffer.put(it) })
+    byteBuffer.flip()
+    val contentOfBody: String = Charsets.UTF_8.decode(byteBuffer).toString()
+
+    assertThat(contentOfBody).isEqualTo(contentOfBody)
+
```

Comment:
😢 A shame that these were not already integration tests - they would have just worked with the asnyc client I think